src/services/storage.js

```diff
@@ -17,135 +17,119 @@
     with this program; if not, write to the Free Software Foundation, Inc.,
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
-import PouchDB from 'pouchdb-react-native';
+import AsyncStorage from '@react-native-community/async-storage';
 import merge from 'lodash.merge';
 import { Sentry } from 'react-native-sentry';
+import PouchDBStorage from './pouchDBStorage';
 
-function Storage(name: string, opts: ?Object = {}) {
+const STORAGE_SETTINGS_KEY = 'storageSettings';
+
+function Storage(name: string) {
   this.name = name;
-  this.opts = { ...opts };
-  this.connect();
+  this.prefix = `wallet-storage:${this.name}:`;
 }
 
-Storage.prototype.connect = function () {
-  this.needToReconnect = false;
-  this.connection = new PouchDB(this.name, this.opts);
-  console.log(`Connected to the database ${this.name}`);
-};
-
-Storage.prototype.db = function () {
-  if (this.needToReconnect) this.connect();
-  return this.connection;
+Storage.prototype.getKey = function (id: string) {
+  return this.prefix + id;
 };
 
-Storage.prototype.get = function (id: string) {
-  return this.db().get(id).catch(() => ({}));
+Storage.prototype.get = async function (id: string) {
+  const data = await AsyncStorage.getItem(this.getKey(id))
+    .then(JSON.parse)
+    .catch(() => {});
+  return data || {};
 };
 
-Storage.prototype.getConflicts = function (): Promise<String[]> {
-  return this.getAllDocs()
-    .then(({ rows }) => (
-      rows.map(({ doc }) => doc._conflicts).reduce((memo, item) => memo.concat(item), [])
-    ));
-};
+const activeDocs = {};
+Storage.prototype.save = async function (id: string, data: Object, forceRewrite: boolean = false) {
+  const currentValue = await this.get(id);
+  const key = this.getKey(id);
+
+  if (activeDocs[key]) {
+    Sentry.captureMessage('Race condition spotted', {
+      extra: {
+        id,
+        data,
+        forceRewrite,
+      },
+    });
```

Comment:
What should happen after this? Just carry on?

In other hand, in development we won't notice anything happened, right? :thinking:

Reply:
It's a primitive check for a race condition. Once we get the error report in sentry - we should review the algorithm.
> In other hand, in development we won't notice anything happened, right? 🤔
As it usually happens, you don't get this kind of errors in development :D plus we have a test for that case